### PR TITLE
layers: Fix discard rectangle draw VU

### PR DESCRIPTION
**Pull request by spencer-lunarg**
Have internal request to fix the draw time check for dynamic discard rectangle so it checks all indexes are set

https://gitlab.khronos.org/vulkan/vulkan/-/merge_requests/5625

### CODE REVIEW

File: tests/vklayertests_others.cpp [hunk 1]
```diff
@@ -11674,11 +11674,12 @@ TEST_F(VkLayerTest, ValidateDiscardRectanglesDynamicStateNotSet) {
         GTEST_SKIP() << "Test not supported by MockICD";
     }
 
-    const VkRect2D rect = {{0, 0}, {16, 16}};
+    auto vkCmdSetDiscardRectangleEXT =
```
Comment (arno-lunarg) [inline tests/vklayertests_others.cpp:11677]:
```suggestion
    auto vkCmdSetDiscardRectangleEXT = GetDeviceProcAddr<PFN_vkCmdSetDiscardRectangleEXT>("vkCmdSetDiscardRectangleEXT")
```

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -490,6 +489,22 @@ bool CoreChecks::ValidateDrawDynamicState(const CMD_BUFFER_STATE &cb_state, cons
         }
     }
 
+    // vkCmdSetDiscardRectangleEXT needs to be set on each rectangle
+    const auto *discard_rectangle_state = LvlFindInChain<VkPipelineDiscardRectangleStateCreateInfoEXT>(pipeline.PNext());
+    if (discard_rectangle_state) {
+        for (uint32_t i = 0; i < discard_rectangle_state->discardRectangleCount; i++) {
+            if (!cb_state.dynamic_state_value.discard_rectangles.test(i)) {
+                LogObjectList objlist(cb_state.commandBuffer());
```
Comment (arno-lunarg) [inline layers/core_validation.cpp:497]:
```suggestion
                const LogObjectList objlist(cb_state.commandBuffer(), pipeline.pipeline());
```

**Review by arno-lunarg — APPROVED** [review]
LGTM! Just 2 code suggestions

File: tests/vklayertests_others.cpp (hunk 1 above)
Comment (spencer-lunarg) [reply tests/vklayertests_others.cpp:11677]: thanks, I need to scrub all these old style ones out still I guess

Comment (arno-lunarg) [reply tests/vklayertests_others.cpp:11676]: The improvement you suggested when we last talked about this problem of getting function pointers would be the best answer to this code flow. I think it was something like:
```
GET_DEVICE_PROC_ADDR(vkCmdSetDiscardRectangleEXT);
```
with the macro handling the get_proc_addr call and variable definition.

Comment (jeremyg-lunarg) [reply tests/vklayertests_others.cpp:11676]: Is this going to get fixed in a re-push or are you planning a global scrub PR?

**Review by jeremyg-lunarg — APPROVED** [review]
LGTM

File: tests/vklayertests_others.cpp (hunk 1 above)
Comment (spencer-lunarg) [reply tests/vklayertests_others.cpp:11676]: We talked about this internally, I forgot if we created an issue, but I will not get to it anytime soon, but I did a global scrub on the `LogObjectList` (the other comment Arno pointed out)

**Comment by spencer-lunarg** [comment]
The WG agreed this was correct

**Pull request merged**